components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,299 @@
+// Package wheeledodometry implements an odometery estimate from an encoder wheeled base
+package wheeledodometry
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/utils"
+)
+
+var (
+	model         = resource.DefaultModelFamily.WithModel("wheeled-odometry")
+	lastLeft      = 0.0
+	lastRight     = 0.0
+	lastTheta     = 0.0
+	distanceLeft  = 0
+	distanceRight = 0
+)
+
+// Config is used for converting a wheeledodometry MovementSensor config attributes.
+type Config struct {
+	LeftMotors   []string `json:"left_motors"`
+	RightMotors  []string `json:"right_motors"`
+	Base         string   `json:"base"`
+	TimeInterval float64  `json:"time_interval,omitempty"`
+}
+
+type odometry struct {
+	resource.Named
+	resource.AlwaysRebuild
+
+	lastLeftPos        float64
+	lastRightPos       float64
+	baseWidth          float64
+	wheelCircumference float64
+	timeInterval       float64
+
+	leftMotors  []motor.Motor
+	rightMotors []motor.Motor
+
+	mmPerRevolution []float64
+
+	angularVelocity spatialmath.AngularVelocity
+	acceleration    r3.Vector
+	linearVelocity  r3.Vector
+	position        r3.Vector
+	orientation     spatialmath.EulerAngles
+
+	cancelFunc              func()
+	activeBackgroundWorkers sync.WaitGroup
+	mu                      sync.Mutex
+	logger                  golog.Logger
+}
+
+func init() {
+	resource.RegisterComponent(
+		movementsensor.API,
+		model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{
+			Constructor: func(
+				ctx context.Context,
+				deps resource.Dependencies,
+				conf resource.Config,
+				logger golog.Logger,
+			) (movementsensor.MovementSensor, error) {
+				return NewWheeledOdometry(ctx, deps, conf, logger)
```

Comment:
make the constructor have the form 
```go
Cosntructor: newWheeledOdometry
```

so remove the function declaration in the constructor, like what wheeled base has now in its constructor function